Accept config options "cert" and "key".

peichman-umd opened the pull request:
These values are used to set the cert tuple for requests. In the SimpleHTTPResolver, you may have one `cert` and one `key` configured. In the TemplateHTTPResolver, you may have a default `cert` and `key` configured in the `[resolver]` config section, and you may override this default on a per-template basis by putting a `cert` and `key` in the appropriate `[[template-name]]` subsection.

peichman-umd committed:
Accept config options "cert" and "key".

These values are used to set the cert tuple for requests. In the SimpleHTTPResolver, you may have one cert and one key configured. In the TemplateHTTPResolver, you may have a default cert and key configured in the [resolver] config section, and you may override this default on a per-template basis by putting a cert and key in the appropriate [[template-name]] subsection.

bcail commented on etc/loris2.conf:
```diff
@@ -65,6 +65,8 @@ src_img_root = '/usr/local/share/images' # r--
 #user='<if needed else remove this line>'
 #pw='<if needed else remove this line>'
+#cert='<SSL client cert for authentication>'
+#key='<SSL client key for authentication>'
```
I'm not a Loris committer, but would it make sense to make these commented values the same as the commented values below in the optional overrides?

jpstroop replied:
I think the whole thing could use some cleanup, but--unless I'm missing something--I think the PR as-is matches the existing (albeit somewhat confusing) structure, and I'm :ok:.

peichman-umd commented:
I can see it working either way. By leaving as-is, it parallels the current
structure of the comments with an explanation of the settings in the main
section with the overrides having actual example strings. But if you use
actual example values in all the settings, it may be simpler to read and be
in line with other sample config files that are out there.

On Mon, Oct 24, 2016 at 10:17 AM, bcail notifications@github.com wrote:

> ## _@bcail_ commented on this pull request.
> 
> In etc/loris2.conf
> https://github.com/loris-imageserver/loris/pull/245#pullrequestreview-5458064
> :
> 
> > @@ -65,6 +65,8 @@ src_img_root = '/usr/local/share/images' # r--
> >  #cache_root='/usr/local/share/images/loris'
> >  #user='<if needed else remove this line>'
> >  #pw='<if needed else remove this line>'
> > +#cert='<SSL client cert for authentication>'
> > +#key='<SSL client key for authentication>'
> 
> I'm not a Loris committer, but would it make sense to make these commented
> values the same as the commented values below in the optional overrides?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/loris-imageserver/loris/pull/245#pullrequestreview-5458064,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AIGXsvjsaOLrTvwXnxAuB0blAHvbl-C1ks5q3L39gaJpZM4Ke1Pq
> .

## 

Peter Eichman
Senior Software Developer
University of Maryland Libraries
peichman@umd.edu

jpstroop commented:
@peichman-umd could you update the [docstring](https://github.com/peichman-umd/loris/blob/ea5f694d7f7c521ceeb0cdc8d680cddf65edf38a/loris/resolver.py#L142-L150) to explain these new options?

peichman-umd committed:
Updated docstrings in resolver.py.

Added SSL cert authentication info, the new structure of the TemplateHTTPResolver config, and the ssl_check option.

peichman-umd commented:
@jpstroop Docstring updated as requested.
